lib/engine/share_price.rb

```diff
@@ -21,19 +21,24 @@ def self.from_code(code, row, column)
           :yellow
         end
 
-      SharePrice.new([row, column], price: price, can_par: can_par, color: color)
+      SharePrice.new([row, column], price: price, can_par: can_par, color: color, unlimited_colors: unlimited_colors)
     end
 
-    def initialize(coordinates, price:, can_par: false, color: nil)
+    def initialize(coordinates, price:, can_par: false, color: nil, unlimited_colors: [])
       @coordinates = coordinates
       @price = price
       @color = color
       @can_par = can_par
       @corporations = []
+      @unlimited_colors = unlimited_colors
     end
 
     def id
       "#{@price},#{@coordinates.join(',')}"
     end
+
+    def counts_for_cert_limit
```

Comment:
minor, but maybe call it 

counts_for_limit?
or 
counts_for_cert_limit?

order is my preference